pkg/controller/hostpathprovisioner/cryptopolicy_test.go

```diff
@@ -0,0 +1,121 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package hostpathprovisioner
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	promv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	ocpconfigv1 "github.com/openshift/api/config/v1"
+	secv1 "github.com/openshift/api/security/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+
+	hppv1 "kubevirt.io/hostpath-provisioner-operator/pkg/apis/hostpathprovisioner/v1beta1"
+	"kubevirt.io/hostpath-provisioner-operator/version"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Controller reconcile loop", func() {
+	Context("crypto policy", func() {
+		BeforeEach(func() {
+			watchNamespaceFunc = func() (string, error) {
+				return testNamespace, nil
+			}
+			version.VersionStringFunc = func() (string, error) {
+				return versionString, nil
+			}
+		})
+
+		It("Should respect cluster-wide crypto config", func() {
+			req := reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      "test-name",
+					Namespace: testNamespace,
+				},
+			}
+			apiServer := &ocpconfigv1.APIServer{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Spec: ocpconfigv1.APIServerSpec{
+					TLSSecurityProfile: &ocpconfigv1.TLSSecurityProfile{
+						Type:   ocpconfigv1.TLSProfileModernType,
+						Modern: &ocpconfigv1.ModernTLSProfile{},
+					},
+				},
+			}
+			nn := client.ObjectKeyFromObject(apiServer)
+			// Register operator types with the runtime scheme.
+			s := scheme.Scheme
+			s.AddKnownTypes(hppv1.SchemeGroupVersion, &hppv1.HostPathProvisioner{})
+			s.AddKnownTypes(hppv1.SchemeGroupVersion, &hppv1.HostPathProvisionerList{})
+			promv1.AddToScheme(s)
+			secv1.Install(s)
+			ocpconfigv1.Install(s)
+
+			// Create a fake client to mock API calls.
+			cl := erroringFakeCtrlRuntimeClient{
+				Client: fake.NewFakeClientWithScheme(s, apiServer),
+				errMsg: "",
+			}
+
+			// Create a ReconcileMemcached object with the scheme and fake client.
+			r := &ReconcileHostPathProvisioner{
+				client:   cl,
+				scheme:   s,
+				recorder: record.NewFakeRecorder(250),
+				Log:      logf.Log.WithName("hostpath-provisioner-operator-controller-test"),
+			}
+			res, err := r.Reconcile(context.TODO(), req)
```

Comment:
I don't think you need to run the reconcile loop at all. just call handleApiServerFunc and ensure the env variables are what you expect them to be. Maybe you are calling them to ensure the handleApiServerFunc doesn't mess with the reconcile loop? But shouldn't you call the function before the reconcile loop then?

Reply:
Yep, you're right, no need for a reconciler struct at all, we'd be testing ctrl runtime primitives that way
Removed

WRT to the empty util -> I prefer we keep it, seems like a nice step towards moving stuff there